RiotSwiftUI/Modules/Settings/ChangePassword/View/ChangePasswordScreen.swift

```diff
@@ -0,0 +1,136 @@
+// limitations under the License.
+//
+
+import SwiftUI
+
+struct ChangePasswordScreen: View {
+
+    // MARK: - Properties
+    
+    // MARK: Private
+    
+    @Environment(\.theme) private var theme
+
+    enum Field { case oldPassword, newPassword1, newPassword2 }
+    @State private var focusedField: Field?
+    
+    // MARK: Public
+    
+    @ObservedObject var viewModel: ChangePasswordViewModel.Context
+    
+    // MARK: Views
+    
+    var body: some View {
+        ScrollView {
+            VStack(spacing: 0) {
+                header
+                    .padding(.top, 16)
+                    .padding(.bottom, 36)
+                form
+            }
+            .readableFrame()
+            .padding(.horizontal, 16)
+        }
+        .background(theme.colors.background.ignoresSafeArea())
+        .alert(item: $viewModel.alertInfo) { $0.alert }
+        .accentColor(theme.colors.accent)
+    }
+
+    /// The title and icon at the top of the screen.
+    var header: some View {
+        VStack(spacing: 8) {
+            OnboardingIconImage(image: Asset.Images.authenticationPasswordIcon)
+                .padding(.bottom, 16)
+
+            Text(VectorL10n.settingsChangePassword)
+                .font(theme.fonts.title2B)
+                .multilineTextAlignment(.center)
+                .foregroundColor(theme.colors.primaryContent)
+                .accessibilityIdentifier("titleLabel")
+        }
+    }
+
+    /// The text fields and submit button.
+    var form: some View {
+        VStack(alignment: .leading, spacing: 12) {
+            RoundedBorderTextField(placeHolder: VectorL10n.settingsOldPassword,
+                                   text: $viewModel.oldPassword,
+                                   isFirstResponder: focusedField == .oldPassword,
+                                   configuration: UIKitTextInputConfiguration(returnKeyType: .next,
+                                                                              isSecureTextEntry: true),
+                                   onCommit: { focusedField = .newPassword1 })
+            .accessibilityIdentifier("oldPasswordTextField")
+
+            RoundedBorderTextField(placeHolder: VectorL10n.settingsNewPassword,
+                                   text: $viewModel.newPassword1,
+                                   isFirstResponder: focusedField == .newPassword1,
+                                   configuration: UIKitTextInputConfiguration(returnKeyType: .next,
+                                                                              isSecureTextEntry: true),
+                                   onCommit: { focusedField = .newPassword2 })
+            .accessibilityIdentifier("newPasswordTextField1")
+
+            RoundedBorderTextField(placeHolder: VectorL10n.settingsConfirmPassword,
+                                   text: $viewModel.newPassword2,
+                                   isFirstResponder: focusedField == .newPassword2,
```

Comment:
Totally optional, just thinking out loud: it might be nice to use `isError:` and `footer:` to show an error when the passwords don't match. Although maybe only validate this in `submit` so the second field doesn't immediately go into an error when you start typing in the first field.

Reply:
I've updated `canSubmit` and checked if passwords match on submission.